### DISCUSSION TITLE
Option to add watermark to image loaded using Image component

### DISCUSSION THREAD

**Discussion by manishkumar03** [vercel/next.js#55925] [Ideas]
### Goals

1. Add a custom watermark to images after it has loaded in the `Image` component
2. If it cannot be done directly in the `Image` component, allow for a callback so that another library can be used for this purpose.

### Non-Goals

1.
2.
3.

### Background

I want to display images on my SaaS website but I want them to be watermarked unless the user has paid for the annual subscription.

### Proposal

No idea how this can be implemented and unfortunately I won't be able to contribute towards it.

**Comment by TusharRedlioDesigns** [comment]
I need same functionality.
